flink-tests/src/test/java/org/apache/flink/test/checkpointing/EventTimeWindowCheckpointingITCase.java

```diff
@@ -91,7 +91,7 @@
 
 	private TestingServer zkServer;
 
-	public MiniClusterResource miniClusterResource;
```

Comment:
a simple MiniClusterResource would be sufficient in this test, and probably others as well. I think it makes sense to do this as part of the separation.

Reply:
You're right. I'll iterate over all the `MiniClusterWithClientResource` and replace them with `MiniClusterResource` where possible.

Reply:
I think in this case, we still need the `MiniClusterWithClientResource` because we rely on the `TestStreamEnvironment` being set.

Reply:
I could only change `MiniClusterWithClientResource` into `MiniCluster` for the following tests `RollingSinkITCase`, `RollingSinkSecuredITCase`, `JarRunHandlerTest`, `SystemResourcesMetricsITCase`, `JobRetrievalITCase`, `BigUserProgramJobSubmitITCase`. All other tests actually require the `MiniClusterWithClientResource`.